### PR TITLE
Integration of Hashicorp Vault with Kubernetes

### PR DESCRIPTION
**Pull request by erzetpe**

### CODE REVIEW

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    exit_with_error "No options passed to script. Aborting.";
+fi
+
+source "$CONFIG_FILE";
+
+INIT_FILE_PATH="$VAULT_INSTALL_PATH/init.txt"
+VAULT_CONFIG_DATA_PATH="$VAULT_INSTALL_PATH/config"
+export VAULT_ADDR="http://$VAULT_IP:8200"
+export KUBECONFIG=/etc/kubernetes/admin.conf
+PATH=$VAULT_INSTALL_PATH/bin:/usr/local/bin/:$PATH
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh:255]: The PATH variable is not exported, is that intended?

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+initialize_vault "$INIT_FILE_PATH";
+
+if [ "${SCRIPT_AUTO_UNSEAL,,}" = "true" ] ; then
+    unseal_vault "$INIT_FILE_PATH";
+fi
+
+check_if_vault_is_unsealed;
+
+log_and_print "Logging into Vault.";
+LOGIN_TOKEN="$(grep "Initial Root Token:" "$INIT_FILE_PATH" | awk -F'[ ]' '{print $4}')";
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh:270]: What if this command fails and token is undefined? Wouldn't this be better to check if before running the vault binary?

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml [hunk 3]
```diff
@@ -116,51 +122,85 @@
...
+
+    - name: Create vault endpoint configuration on kubernetes cluster
+      template:
+        dest: "{{ specification.vault_install_dir }}/kubernetes/vault-endpoint-configuration.yml"
+        owner: "{{ specification.vault_system_user }}"
+        group: "{{ specification.vault_system_group }}"
+        mode: u=rw,g=r,o=
+        src: kubernetes/vault-endpoint-configuration.yml.j2
+
+    - name: Create Vault service account
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml:176]:
```suggestion
    - name: Create Vault service account configuration
```

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml [hunk 4]
```diff
@@ -116,51 +122,85 @@
...
+      template:
+        dest: "{{ specification.vault_install_dir }}/kubernetes/vault-service-account.yml"
+        owner: "{{ specification.vault_system_user }}"
+        group: "{{ specification.vault_system_group }}"
+        mode: u=rw,g=r,o=
+        src: kubernetes/vault-service-account.yml.j2
+
+    - name: Create application service account
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml:184]:
```suggestion
    - name: Create application service account configuration
```

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml [hunk 5]
```diff
@@ -116,51 +122,85 @@
...
+      template:
+        dest: "{{ specification.vault_install_dir }}/kubernetes/app-service-account.yml"
+        owner: "{{ specification.vault_system_user }}"
+        group: "{{ specification.vault_system_group }}"
+        mode: u=rw,g=r,o=
+        src: kubernetes/app-service-account.yml.j2
 
     - name: Run configuration script
+      #become: yes
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml:193]: Is the `become: yes` needed here then?

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/kubernetes/app-service-account.yml.j2 [hunk 6]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/kubernetes/app-service-account.yml.j2:1]: Please add the "ansible_managed" stuff here.

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/kubernetes/vault-endpoint-configuration.yml.j2 [hunk 7]
```diff
@@ -0,0 +1,20 @@
+---
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/kubernetes/vault-endpoint-configuration.yml.j2:1]: Please add the "ansible_managed" stuff here.

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/kubernetes/vault-service-account.yml.j2 [hunk 8]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/kubernetes/vault-service-account.yml.j2:1]: Please add the "ansible_managed" stuff here.

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/vault/policy-application.hcl.j2 [hunk 9]
```diff
@@ -0,0 +1,6 @@
+path "{{ specification.secret_mount_path }}/data/{{ specification.app_secret_path }}/*" {
```
Comment (sk4zuzu) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/templates/vault/policy-application.hcl.j2:1]: Please add the "ansible_managed" stuff here.

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml (hunk 5 above)
Comment (erzetpe) [reply core/src/epicli/data/common/ansible/playbooks/roles/vault/tasks/main.yml:193]: No. :)

File: core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+# TODO: Configuration of unsealing after server restart with script
+# TODO: Add configurable log paths
+
+HELP_MESSAGE="Usage: configure-vault.sh -c SCRIPT_CONFIGURATION_FILE_PATH -a VAULT_IP_ADDRESS"
+
+function print_help { echo "$HELP_MESSAGE"; }
+
+function log_and_print {
+    local string_to_log="$1";
+    echo "`date +"%Y-%m-%d-%H:%M:%S"` - $string_to_log" | tee -a /opt/vault/logs/configure_vault.log;
```
Comment (ar3ndt) [inline core/src/epicli/data/common/ansible/playbooks/roles/vault/files/vault/configure-vault.sh:14]: Can we consider using $() instead of `` ?

**Pull request closed without merging**